task/oracle-cloud-storage-upload/0.1/oracle-cloud-storage-upload.yaml

```diff
@@ -0,0 +1,501 @@
+        NAMESPACE=$(cat "$NAMESPACE_FILE")
+        log "Using namespace from credentials secret"
+      elif [[ -z "$NAMESPACE" ]]; then
+        log "No namespace provided, retrieving automatically..."
+
+        log "Attempting to detect Object Storage namespace..."
+        if NAMESPACE=$(timeout_cmd 60 oci os ns get --query 'data' --raw-output 2>/dev/null); then
+          log "Using auto-detected namespace: $NAMESPACE"
+        else
+          exit_code=$?
+          if [ $exit_code -eq 124 ]; then
+            log "ERROR: Namespace detection timed out after 60 seconds"
+          else
+            log "ERROR: Failed to detect Object Storage namespace"
+          fi
+          log "Please provide namespace parameter or add it to credentials secret"
+          exit $exit_code
+        fi
+      else
+        log "Using namespace from parameter"
+      fi
+
+      log "OCI Config prepared for region=$REGION, namespace=$NAMESPACE"
+
+      # Validate compatibility of deleteExtraFiles and replaceExistingFiles
+      if [[ "$DELETE_EXTRA_FILES" == "true" && "$REPLACE_EXISTING_FILES" == "false" ]]; then
+        log "ERROR: deleteExtraFiles can only be used when replaceExistingFiles is true"
+        exit 1
+      fi
+
+      # Check source path
+      if [[ ! -e "$SOURCE_PATH" ]]; then
+        log "ERROR: Source path $SOURCE_PATH does not exist"
+        exit 1
+      fi
+
+      log "Uploading from $SOURCE_PATH to bucket $BUCKET_NAME (namespace: $NAMESPACE, prefix: $OBJECT_PREFIX)"
+
+      # Extra arguments are passed as script arguments via Tekton args
+      EXTRA_ARGS_ARRAY=("${@}")
+      log "Expanded extra arguments: ${EXTRA_ARGS_ARRAY[*]}"
+      if [[ ${#EXTRA_ARGS_ARRAY[@]} -gt 0 ]]; then
+        log "Using extra arguments: ${EXTRA_ARGS_ARRAY[*]}"
+      fi
+
+      if [[ -d "$SOURCE_PATH" ]]; then
+        # Directory upload
+        log "Starting directory upload..."
+        if [[ "$DELETE_EXTRA_FILES" == "true" ]]; then
+          SYNC_PARAMS="--delete"
+          if [[ -n "$OBJECT_PREFIX" ]]; then
+            SYNC_PARAMS="$SYNC_PARAMS --prefix $OBJECT_PREFIX"
+          fi
+          log "Running sync with delete option (timeout: 30 minutes)..."
+          if [[ ${#EXTRA_ARGS_ARRAY[@]} -gt 0 ]]; then
+            if timeout_cmd 1800 oci os object sync --bucket-name "$BUCKET_NAME" --namespace "$NAMESPACE" \
+              --src-dir "$SOURCE_PATH" $SYNC_PARAMS "${EXTRA_ARGS_ARRAY[@]}"; then
+              log "Directory sync completed successfully"
+            else
+              exit_code=$?
+              if [ $exit_code -eq 124 ]; then
+                log "ERROR: Directory sync operation timed out after 30 minutes"
+              else
+                log "ERROR: Directory sync operation failed"
+              fi
+              exit $exit_code
+            fi
+          else
+            if timeout_cmd 1800 oci os object sync --bucket-name "$BUCKET_NAME" --namespace "$NAMESPACE" \
```

Comment:
This takes away from users the ability to set a timeout. Tekton already has an API for timeouts, if we enforce a timeout here the Tekton timeout will only trigger if it's shorter then this arbitrary value. This time depends very much on the size of what's being uploaded.